src/scrot.c

```diff
@@ -411,24 +410,20 @@ void scrotGrabMousePointer(const Imlib_Image image, const int xOffset,
     const unsigned short height = xcim->height;
     const int x = (xcim->x - xcim->xhot) - xOffset;
     const int y = (xcim->y - xcim->yhot) - yOffset;
-    DATA32 *pixels = NULL;
-
-#ifdef __i386__
-    pixels = (DATA32 *)xcim->pixels;
-#else
-    DATA32 data[width * height * 4];
-
-    unsigned int i;
+    DATA32 *pixels;
+    size_t i;
+
+    if ((pixels = reallocarray(NULL, width * height, sizeof(pixels))) == NULL)
+        err(EXIT_FAILURE, "reallocarray");
```

Comment:
Before exiting you must call `XFree(xcim)` because it is memory allocated by a library and we have no control over it.

Reply:
The process' memory is deallocated when it exits.